mobile/status_request_log.go

```diff
@@ -0,0 +1,51 @@
+package statusgo
+
+import (
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/status-im/status-go/logutils"
+)
+
+// logAndCall logs request call details and executes the fn function if logging is enabled
+func logAndCall[T any](method string, params any, fn func() (T, error)) (T, error) {
+	var startTime time.Time
+	var duration time.Duration
+
+	if logutils.IsRequestLoggingEnabled() {
+		startTime = time.Now()
+	}
+
+	resp, err := fn()
+
+	if logutils.IsRequestLoggingEnabled() {
+		duration = time.Since(startTime)
+		logutils.GetRequestLogger().Debug(method, "params", fmt.Sprintf("%+v", params), "resp", fmt.Sprintf("%+v", resp), "err", err, "duration", duration)
+	}
+
+	return resp, err
+}
+
+func logAndCallString(method string, params any, fn func() string) string {
+	resp, _ := logAndCall(method, params, func() (string, error) {
+		return fn(), nil
+	})
+	return resp
+}
+
+func logAndCallVoid(method string, params any, fn func()) {
+	_, _ = logAndCall(method, params, func() (any, error) {
+		fn()
+		return nil, nil
+	})
+}
```

Comment:
I'm not a fan of such special functions, which in fact do the same thing.
We could create a `WithResponse` 

We could make it like this instead:
```go
type Option func() []interface{}

func WithResponse(response string) Option {
	return func() []interface{} {
		return []interface{}{"response", response}
	}
}

func NewCallLogger(method string, request string) *CallLogger {
	return &CallLogger{
		method:    method,
		request:   request,
		startTime: time.Now(),
	}
}

func (cl *CallLogger) end(opts ...Option) {
	params := []interface{}{
		"duration", time.Since(cl.startTime),
		"request", cl.request,
	}
	for _, opt := range opts {
		params = append(params, opt()...)
	}
	log.Debug(cl.method, params...)
}
```

And use like this:
```go
	cl := NewCallLogger("Foo", requestJSON)
	defer func() { cl.end(WithResponse(responseJSON)) }()
```

It's not that convenient as direct argument, but more Go-style.
I don't insist on this, we can go with the current approach as well.